### PR TITLE
Cam guided training

### PR DESCRIPTION
**Pull request by JoeCx**
# Overview

**Type of Change:** New Feature

**Summary:** Created a CAM(class activationmapping) Training program class(CAMGuidedTrainingProgram). This uses masks to further calculate the loss and penalize the model when looking outside of the targeted area. Additionally, I fixed the error in the main training program when creating the train/test datasets for training. We were applying the train_transformations to both datasets, but now we are applying the train transformations to the train dataset, and the normal transformations to the test dataset.

## UI/UX Implementation

No changes made.

## Model Updates

This section describes changes that were made to the models used by your application.

### Data Models

No changes made.

### Object-Oriented Models

No changes made.

## End-to-End Testing Instructions

Run the simulator files for the original training program to see the difference after the train/test fix. Hyperparameter training needs to be rerun.

### CODE REVIEW

**Review by tmckeown311 — APPROVED** [review]
Tested the training program and found everything working as intended. Models train and save to the correct locations and work well!
